File: src_fortran/sub/freeunit.f
c
c
c
c
c
c
      subroutine FREEUNIT(iunit)
      implicit none
c
c     Purpose: Get a free FORTRAN-unit
c
c     Output: iunit  a free (file-)unit number
c
c     Urs Kradolfer, 16.7.90
c
      integer iunit
      logical lopen
c
      do iunit=10,999
         if(iunit.eq.999) stop'FREEUNIT>>> no free unit found!'
         inquire(unit=iunit,opened=lopen)
         if(.not.lopen) RETURN
      enddo
c
      RETURN
c
      end ! of subr. freeunit

File: src/sub/freeunit.c
#include <stdio.h>
#include <stddef.h>

void freeunit(int *iunit) {
    int i;
    FILE *f;

    if (iunit == NULL) {
        return;
    }

    for (i = 10; i < 999; ++i) {
        f = fdopen(i, "r");
        if (f == NULL) {
            *iunit = i;
            return;
        }
    }

    *iunit = 10;
}
